e2e/cypress/integration/enterprise/extend_session/email_login_activity_spec.js

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import users from '../../../fixtures/users.json';
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @enterprise @extend_session
+
+describe('Extend Session - Email Login', () => {
+    const townSquarePage = '/ad-1/channels/town-square';
+    const oneDay = 24 * 60 * 60 * 1000;
+    const testUser = 'user-1';
+
+    before(() => {
+        // # Login as sysadmin and check if with license and has matching database
+        cy.apiLogin('sysadmin');
+        cy.requireLicense();
+        cy.requireServerDBToMatch();
+    });
+
+    beforeEach(() => {
+        // # Login as sysadmin and revoke sessions of the test user
+        cy.apiLogin('sysadmin');
+        cy.dbGetUser({username: testUser}).then(({user}) => {
+            cy.apiRevokeUserSessions(user.id);
+        });
+    });
+
+    it('should redirect to login page when session expired', () => {
+        // # Update system config
+        const setting = {
+            ServiceSettings: {
+                ExtendSessionLengthWithActivity: true,
+                SessionLengthWebInDays: 1,
+            },
+        };
+        cy.apiUpdateConfig(setting);
+
+        // # Login as test user and go to town-square channel
+        cy.apiLogin(testUser);
+        cy.visit(townSquarePage);
+
+        // # Get active user sessions as baseline reference
+        cy.dbGetActiveUserSessions({username: testUser}).then(({sessions: initialSessions}) => {
+            // Post a message to a channel
+            cy.postMessage(Date.now());
+
+            const expiredSession = parseDateTime(initialSessions[0].createat, 10) + 1;
```

Comment:
I'm confused here how adding 1ms to existing ExpiresAt necessarily expires that session.  If the current ExpiresAt is past then it is already expired, and if not, adding 1ms is not likely to make it so.  I might be missing something.

Reply:
ExpiresAt here is still in the future which is on next day after a session is created on [login](https://github.com/mattermost/mattermost-webapp/pull/5630/files#diff-6298b162b7c6ee045aebc986b74f5489R45). I picked up the lowest value of 1ms to guarantee that it's expired during the time of [reloading a page](https://github.com/mattermost/mattermost-webapp/pull/5630/files#diff-6298b162b7c6ee045aebc986b74f5489R66).